numpy/doc/structured_arrays.py

```diff
@@ -6,231 +6,506 @@
+structure. The datatype of a field may be any numpy datatype including other
+structured datatypes, and it may also be a :term:`sub-array` which behaves like
+an ndarray of a specified shape. The offsets of the fields are arbitrary, and
+fields may even overlap. These offsets are usually determined automatically by
+numpy but can also be manually specified.
+
+Structured Datatype Creation
+----------------------------
+
+Structured datatypes may be created using the function :func:`numpy.dtype` with
+one of 4 alternative forms of specification which vary in flexibility and
+conciseness. The forms of specification are further documented in the
+:ref:`Data Type Objects <arrays.dtypes.constructing>` reference page, and in
+summary they are:
+
+1. A list of tuples, one tuple per field
+````````````````````````````````````````
+
+Each tuple has the form ``(fieldname, datatype, shape)`` where shape is
+optional. ``fieldname`` is a string (or tuple if titles are used, see
+:ref:`Field Titles <titles>` below), ``datatype`` may be any object convertible
+to a datatype, and shape (optional) is a tuple of integers specifying subarray
+shape.
+
+ >>> np.dtype([('x', 'f4'), ('y', np.float32), ('z', 'f4', (2,2))])
+ dtype=[('x', '<f4'), ('y', '<f4'), ('z', '<f4', (2, 2))])
+
+If ``fieldname`` is the empty string (``''``) the field will be given a default
+name of the form ``f#``, where ``#`` is the integer index of the field,
+counting from 0 from the left::
+
+ >>> np.dtype([('x', 'f4'),('', 'i4'),('z', 'i8')])
+ dtype([('x', '<f4'), ('f1', '<i4'), ('z', '<i8')])
+
+The byte offsets of the fields within the structure and the total structure
+itemsize are determined automatically.
+
+2. A string of comma-separated dtype specifications
+```````````````````````````````````````````````````
+
+In this shorthand notation any of the :ref:`string dtype specifications
+<arrays.dtypes.constructing>` may be used in a string, separated by commas. The
+itemsize and byte offsets of the fields are determined automatically, and the
+field names are given the default names ``f0``, ``f1``, etc. ::
+
+ >>> np.dtype('i8,f4,S3')
+ dtype([('f0', '<i8'), ('f1', '<f4'), ('f2', 'S3')])
+ >>> np.dtype('3int8, float32, (2,3)float64')
+ dtype([('f0', 'i1', 3), ('f1', '<f4'), ('f2', '<f8', (2, 3))])
+
+3. A dictionary of field parameter arrays
+`````````````````````````````````````````
+
+This is the most flexible form of specification since it allows control
+over the byte-offsets of the fields and the itemsize of the structure.
+
+The dictionary has two required keys, 'names' and 'formats', and four optional
+keys, 'offsets', 'itemsize', 'aligned' and 'titles'. 'names' and 'formats'
+should respectively correspond to a list of field names and a list of dtype
+specifications of the same length. The optional 'offsets' key must correspond
+to a list of integer byte-offsets of each field within the structure, of the
```

Comment:
"The optional 'offsets' key is a list of integer byte offsets, one for each field within the structure."